src/mlpack/tests/saga_test.cpp

```diff
@@ -55,8 +55,9 @@ BOOST_AUTO_TEST_CASE(SAGALogisticRegressionTest)
 BOOST_AUTO_TEST_CASE(GeneralizedRosenbrockTest)
 {
   // Loop over several variants.
-  for (size_t i = 10; i < 20; i += 5)
+  for (size_t i = 5; i < 20; i += 2)
   {
+    mlpack::math::RandomSeed(std::time(NULL));
```

Comment:
Please do not use RandomSeed inside the test; see https://github.com/mlpack/mlpack/pull/1306 for more information.

Reply:
@zoq the success of SAGA closely depends on the generation of random numbers. Here, I'm setting the seed not to a fixed value but to `time(NULL)` which I think will change with the iteration. SAGA cannot cope up with various variants of Rosenbrock function without the change of `RandomSeed()` Its failing on some variants in the absence of this. But I tried setting the `RandomSeed(time(NULL))` inside SAGA but it didn't helped. Tests are still failing. What should I do?

Reply:
If SAGA does not reliably converge, this suggests to me that there may be a bug or problem---or perhaps the algorithm parameters need to be tuned.  However, if it is like simulated annealing and it is known that sometimes it simply will not converge because the starting point is bad, I would suggest modifying the test so that you run the optimization multiple times and ensure that at least one of those runs is successful.  You can see the simulated annealing test (`sa_test.cpp`) for an example of that.

Reply:
@rcurtin SAGA does not reliably converge only on some of the variants of Rosenbrock functions and that too in the absence of setting the seed. I don't think if there'll be any bug because then the SAGA will not converge at all and wouldn't even pass the `LogisticRegressionTest` I've seen simulated annealing test and I find it suitable for that because `sa` is very sensitive to initialization but I think it shouldn't be the case for SAGA as its a fast gradient based convergence method. The main issue here is that how can we justify SAGA coping up with all the variants if the seed is set and otherwise not? Let us see what @zoq has to say about it